test/main.cpp

```diff
@@ -368,16 +338,21 @@ EXPORT int main(int argc, char const* argv[])
 		if (filter && tests_to_run.count(_g_unit_tests[i].name) == 0)
 			continue;
 
-		std::snprintf(dir, sizeof(dir), "sub_%i", i);
-		std::string unit_dir = combine_path(current_working_directory(), dir);
+		std::string unit_dir = unit_dir_prefix + std::to_string(i);
 		error_code ec;
 		create_directory(unit_dir, ec);
 		if (ec)
 		{
-			std::printf("Failed to create unit directory: %s\n", ec.message().c_str());
+			std::printf("Failed to create unit test directory: %s\n", ec.message().c_str());
 			return 1;
 		}
-		if (!change_dir(dir)) return 1; // every unit test must have separate sandbox
+		change_directory(unit_dir, ec);
+		if (ec)
+		{
+			std::printf("Failed to change unit test directory: %s\n", ec.message().c_str());
+			return 1;
+		}
+		unit_dirs.push_back(unit_dir);
```

Comment:
the directory should be put in here immediately after it's been (successfully) created. imagine you can create the directory, but fail to cd into it. you still want to delete it later.

In fact, I'm kind of thinking that it would be more robust to delete the test directory inside the loop, right after each individual test passes. Would that make sense you think? Then you wouldn't need to store them in a vector.

In addition, the way I think they should be deleted is by having a scope guard delete it (so it happens regardless of early exits, continues and exceptions). Something like this:
``struct dir_guard { std::string dir; ~dir_guard() { error_code ec; remove_all(dir, ec); } } guard = { unit_dir };``